pkg/quota/evaluator/core/pods.go

```diff
@@ -269,12 +284,45 @@ func isTerminating(pod *api.Pod) bool {
 }
 
 // QuotaPod returns true if the pod is eligible to track against a quota
-func QuotaPod(pod *api.Pod) bool {
-	return !(api.PodFailed == pod.Status.Phase || api.PodSucceeded == pod.Status.Phase)
+// A pod is eligible for quota, unless any of the following are true:
+//  - pod has a terminal phase (failed or succeeded)
+//  - pod has been marked for deletion and grace period has expired.
+func QuotaPod(pod *api.Pod, clock clock.Clock) bool {
+	// if pod is terminal, ignore it for quota
+	if api.PodFailed == pod.Status.Phase || api.PodSucceeded == pod.Status.Phase {
```

Comment:
pods have conditions now, right?  Since you're rewriting the function anyway, can we update to conditions.

Reply:
the pod Phase is still the state machine we must follow to know that a pod has reached the end of its life.  the PodCondition is not sufficient for that need.

Reply:
> the pod Phase is still the state machine we must follow to know that a pod has reached the end of its life. the PodCondition is not sufficient for that need.

This is really weird.  It's been over two years since we decided that conditions were a better choice than phases.  When is the kubelet going to update?

Reply:
Never because of backwards API compatibility.

Reply:
> Never because of backwards API compatibility.

Eh, it only skews two releases.  I'd rather have all our code keep as current as compatibility allows so developers aren't trying to remember: well these conditions match these phases.  Cruft like this is super hard to deal with in controllers and utilities.